### ISSUE TITLE
Model evaluate accuracy drops following Save + Load

### ISSUE THREAD

**Issue by duckduck-sys** [tensorflow/tensorflow#52166]
After training EfficientNetB0 on a custom dataset, i want to ensure that the model i save give the same accuracy after i load it:

```
# Evaluate model
print("After training: ")
model.evaluate(dataset)

# Save model
model.save('my_model.hdf5')

# Reload the saved model
new_model = tf.keras.models.load_model('my_model.hdf5')

# Evaluate again
print("After saving and reloading: ")
new_model.evaluate(dataset)
```

Which gives the output :

```
After training : 
1000/1000 [==============================] - 20s 19ms/step - loss: 0.4617 - accuracy: 0.8139
After saving and reloading : 
1000/1000 [==============================] - 20s 18ms/step - loss: 0.5586 - accuracy: 0.7688
```

Batch size 32. The model got much worse after save + reloading

I tried EfficientNetB1, B2, B3 as well, all same issue

Then I tried simply switching to MobileNetV3 and Xception, they both work perfectly fine! The accuracy after Load is identical to that measured before Save...

Whats going on? Using tf 2.6 with python 3.8 and cudnn 8.2

**Comment by duckduck-sys** [comment]
Found this old [issue](https://github.com/tensorflow/tpu/issues/378) that indicated problems with model saving when using Keras+EfficientNet, so i tried switching to the SavedModel pb format instead of HDF5 format, and it works.

I.e. just remove ".hdf5" form the above code, it then saves in .pb instead, which gives the expected identical output after saving+reloading. 

It also gives a warning though: 

/lib/python3.8/site-packages/keras/utils/generic_utils.py:494: CustomMaskWarning: Custom mask layers require a config and must override get_config. When loading, the custom mask layer must be passed to the custom_objects argument

But it doesn't seem to effect accuracy so i suppose it's ok. Solution is that EfficientNet models should never be saved in hdf5 format.

**Comment by sushreebarsa** [comment]
@duckduck-sys Could you please move this issue to closed status if it is resolved for you ? Please have a look at the [link1](https://github.com/tensorflow/tensorflow/issues/42459), [link2 ](https://stackoverflow.com/questions/61493605/keras-model-containing-efficientnet-submodel-cant-load-weights-vgg-works-fine)and [link3](https://www.tensorflow.org/api_docs/python/tf/saved_model) for your reference.Thank you!

**Comment by duckduck-sys** [comment]
ok, thanks!
